src/kademlia/dht_tracker.cpp

```diff
@@ -285,7 +370,11 @@ namespace libtorrent { namespace dht
 	void dht_tracker::get_item(sha1_hash const& target
 		, boost::function<void(item const&)> cb)
 	{
-		m_dht.get_item(target, boost::bind(&get_immutable_item_callback, _1, cb));
+		boost::shared_ptr<bool> item_posted = boost::make_shared<bool>(false);
```

Comment:
single byte heap allocations seem a bit shady (not very efficient). There's no other object this (and the other heap allocated int) could be coalesced with?